vela-templates/definitions/internal/depends-on-app.cue

```diff
@@ -0,0 +1,64 @@
+import (
+	"vela/op"
+	"encoding/yaml"
+)
+
+"depends-on-app": {
+	type: "workflow-step"
+	annotations: {}
+	labels: {}
+	description: "check or install depends-on Application"
+}
+
+template: {
+	dependsOn: op.#Read & {
+		value: {
+			apiVersion: "core.oam.dev/v1beta1"
+			kind:       "Application"
+			metadata: {
+				name:      parameter.name
+				namespace: parameter.namespace
+			}
+		}
+	}
+
+	load: op.#Steps & {
+		if dependsOn.err != _|_ {
+			configMap: op.#Read & {
+				value: {
+					apiVersion: "v1"
+					kind:       "ConfigMap"
+					metadata: {
+						name:      parameter.name
+						namespace: parameter.namespace
+					}
+				}
+			}
+			apply: op.#Apply & {
+				value: {
+					yaml.Unmarshal(configMap.value.data[parameter.name])
+				}
+			}
+		}
+		if dependsOn.err == _|_ {
+			apply: op.#Apply & {
+				value: {
+					dependsOn.value
+				}
+			}
+		}
+	}
+
+	status: load.apply.value.status
+
+	wait: op.#ConditionalWait & {
+		continue: status.status == "running"
```

Comment:
Why is `status.status`.... Shouldn't it rename to `phase`?

Reply:
good suggestion, I will fix it